---
title: "CNA2Subpathway User Guide"
author: "Junwei Han"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    toc: yes
  pdf_document:
    toc: yes
  prettydoc::html_pretty:
    highlight: github
    theme: cayman
vignette: >
  %\VignetteIndexEntry{CNA2Subpathway User Guide} 
  %\usepackage[utf8]{inputenc} 
  %\VignetteEncoding{UTF-8} 
  %\VignetteEngine{knitr::rmarkdown}

---
```{r style, echo=FALSE, results="asis", message=FALSE}
knitr::opts_chunk$set(tidy = FALSE,
                      warning = FALSE,
                      message = FALSE)
```

### <font face="微软雅黑"> Introduce </font>

<font face="微软雅黑"> This vignette illustrates how to easily use the **CNA2Subpathway package**. This vignette illustrates how to easily use the **CNA2Subpathway package**. Here, we propose a network-based calculated method, CNA2Subpathway, to identify dysfunctional subpathways with respect to somatic CNAs in cancer by integrating CNAs data, gene expression data, and subpathway crosstalk network. 

The IDSCNA method consists of three parts:
   
  - 1.Obtin the CNA label. We made a CNA label based on the CNAs profile and the amplified and deleted genes in the GISTIC2 results. We discretize the CNAs profile based on the amplified gene and the deleted gene to a binary matrix. label 1 represents amplification in the amplified gene and deletion in the deleted gene;label 0 vice versa.

  - 2.Calculate the CNA-DEscore. The CNA-DEscore is defined as differentially expressed genes driven by CNAs. We use Student’s t-test method to compute the gene differentially expressed level between CNA labels, and convert the t-test p-value of each gene to z-score.The z-score is defined as CNA-DEscore, and a larger CNA-DEscore indicates the gene regulated by its CNA to a greater extent.

  - 3.Constructing network and randomization. In our method, we fist constructed a Subpathway(SP)-GO bipartite network. We define an edge between a subpathway and a Go term, if they have a common gene, and give the weight of this edge that calculate by the Jaccard index and CNA-DEscore. Next, we made the SP-GO network convert to SP-SP network, similarly, we define an edge between two subpathways, if thet have common biology function, and the edge weights will be larger for pairs of subpathways that relate more to GO and CNA. Then, we use eigenvector centrality measure to calculate how central each subpathwaty is in this network. Finally, the significance of these centrality scores is assessed using a bootstrap-based randomization method.

   This package provides the **getExampleData** function to return example data and environment variables, such as the data of CNAP and so on.</font>

*****
### <font face="微软雅黑">Example 1 : Convert the CNA profile into a binary matrix. </font>

<font face="微软雅黑">  The function **getCNAlabel** be used to discretize the CNA profile to a binary matrix. This function requires the user to input the Copy Number Alteration profile and those amplified and deleted genes, according to the results of GISTIC2 analysis, and the function can define their own dataset list data as input. We selected a part of the Breast cancer the CNA data in the TCGA database. Those example dates are saved into the package environment variables.
  
The commands are as follows: </font>

```{r eval=TRUE}
#
require(CNA2Subpathway)
# obtain Copy Number Alteration profile data
CNAP<-GetExampleData("CNAP")
head(CNAP[1:10,1:10])
# obtion amplified genes
ampGene<-GetExampleData("ampGene")
# obtion deleted genes 
delGene<-GetExampleData("delGene")
# Obtin the CNA label
CNAlabelresult<-getCNAlabel(CNAP,ampGene,delGene)
# view first ten genes and ten sample of the label
head(CNAlabelresult[1:10,1:10])
```

*****

### <font face="微软雅黑"> Example 2 : Calculate the CNA-DEscore. </font>

<font face="微软雅黑"> The function **getCNADEscore** used to calculate CNA-DEscore for each gene, we use Student’s t-test method to compute the gene differentially expressed level between CNA labels. To compare the genes regulated by CNAs at a common level, we convert the t-test p-value of each gene to z-score. This function requires the user input the gene expression profile(GEP) and the CNA label(the result of the function getCNAlabel).

  The commands are as follows: </font>


```{r}
# obtain GEP data
GEP<-GetExampleData("GEP")
# obtain the result of getCNAlabel function
CNAlabelresult<-GetExampleData("CNAlabelresult")
# Calculate the CNA-DEscore
CNADEscore<-getCNADEscore(GEP,CNAlabelresult)
# view first the ten gene CNA-DEscore
head(CNADEscore[1:10,])
```

*****

###<font face="微软雅黑">  Example 3 : Constructing network and randomization. </font>

<font face="微软雅黑">  The function **IDSubPath** used to find significant dysregulate subpathways by the CNA2Subpathway method. According our method, in this function, the user need input 6 variable. Four of six variable can obtain from our example data, those data from KEGG and GO, and the user can change at need also. The "CNADEscore" variable is result of function getCNADEscore. And the last variable is nperm, that representative number of disturbances, usually nperm = 1000 or bigger as statistically significant. 

  The commands are as follows: </font>

```{r}
# load depend package
require(igraph)
# obtain subpathway data 
SubPathwayInfo<-GetExampleData("SubPathwayInfo")
# obtain biological functions data
GoInfo<-GetExampleData("GoInfo")
# obtain some genes shared by a sub-pathway with biological function
GoSubPconGene<-GetExampleData("GoSubPconGene")
# obtain Jaccardscore shared by a sub-pathway with biological function
Jaccardscore<-GetExampleData("Jaccardscore")
# obtain the result of getDEGscore function
CNADEscore<-GetExampleData("CNADEscore")
# run the IDSubPath function
CNA2Subpathwayresult<-IDSubPath(CNADEscore,nperm=100,Subpathway=SubPathwayInfo,Go=GoInfo,Jaccard=Jaccardscore,Go_SubPath_gene=GoSubPconGene)
# get the result of the IDSubPath function #only show
CNA2Subpathwayresult<-GetExampleData("CNA2Subpathwayresult")
# View subpathway rank by fdr value
head(CNA2Subpathwayresult[order(CNA2Subpathwayresult[,"Fdr"]), ][1:10,])

```

*****


### <font face="微软雅黑"> Visualize 1: Plot a DISCOVER mutual exclusivity test. </font>

<font face="微软雅黑">  The function **plotDiscovermap** used to plot mutal exclusivity picture. The user need input two datas: 1.Those intresting and happened CNA genes; 2.The resutl of function getCNAlabel.

  The commands are as follows: </font>

```{r}
# load depend package
require(discover)
# input gene
genes <- c("AKT3","AKT1","CALML5","PIK3CA","PIK3R1","CALM1","CALM3","CALML3")
# get the result of the getCNAlabel function
CNAlabelresult<-GetExampleData("CNAlabelresult")
# plot discover 
discover<-plotDiscovermap(genes,CNAlabelresult)
plot(discover)

```

*****

## <font face="微软雅黑"> Visualize 2: Plot a subpathway network graph when user input subpathwayID.</font>

#####
<font face="微软雅黑">  The function **plotsubpathway** can visualization of a subpathway network map. The user need input the subpathway pathID of interest.
  
  The commands are as follows:  </font>

```{r}
# load depend package
require(igraph)
# obtain subpathwaymap data
SubPathwaymapdata<-GetExampleData("SubPathwaymapdata")
# plot network graph of the subpathway 04915_13
plotSubPathway("04915_13",SubPathwaymapdata)

```




